### ISSUE TITLE
Failed to minify the code from this file

### ISSUE THREAD

**Issue by nerestaren** [aframevr/aframe#3449]
**Description:**

- A-Frame Version: 0.8.0
- Platform / Device: aframe + create-react-app
- Reproducible Code Snippet or URL: N/A

When upgrading to 0.8.1 or 0.8.0 I cannot build the project anymore.

Apparently, [there are ES6 features in the NPM module](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#npm-run-build-fails-to-minify).

> Creating an optimized production build...
> Failed to compile.
> 
> Failed to minify the code from this file:
> 
>         ./node_modules/aframe/dist/aframe-master.js:75038
> 
> Read more here: http://bit.ly/2tRViJ9

This is the relevant code of aframe-master:

```
  // Give all functions their proper name.
  for (var propName of Object.getOwnPropertyNames(newObj.prototype)) {
    var propVal = newObj.prototype[propName];
    if (typeof propVal === 'function') {
      propVal.displayName = propName;
    }
  }
```

Thank you

**Comment by kromkens** [comment]
I get the same build errors.

- A-Frame Version: 0.8.1
- Platform / Device: A-Frame + create-react-app

**Comment by machenmusik** [comment]
`for ... of` is ES6

**Comment by machenmusik** [comment]
That seems to have introduced by https://github.com/aframevr/aframe/commit/f5ef24b3ef1537d02bf653ea0e131cdfa37f4285

**Comment by walaszczykm** [comment]
In the project I'm working on this error also occurred. I checked the line 62 in a-register-element.js which contains this relevant code part. This is the only part which causes build error while using react-scripts.

**Comment by walaszczykm** [comment]
I prepared fix for this issue and create PR #3495

**Comment by machenmusik** [comment]
nice, thank you!
